packages/swapper/src/swappers/JupiterSwapper/endpoints.ts

```diff
@@ -0,0 +1,144 @@
+import { CHAIN_NAMESPACE, fromAssetId, solAssetId } from '@shapeshiftoss/caip'
+import type { BuildSendApiTxInput, GetFeeDataInput } from '@shapeshiftoss/chain-adapters'
+import type { SolanaSignTx } from '@shapeshiftoss/hdwallet-core'
+import type { KnownChainIds } from '@shapeshiftoss/types'
+import { PublicKey, type TransactionInstruction } from '@solana/web3.js'
+import type { AxiosError } from 'axios'
+
+import type { GetUnsignedSolanaTransactionArgs } from '../../types'
+import { type SwapperApi } from '../../types'
+import { checkSolanaSwapStatus, isExecutableTradeQuote, isExecutableTradeStep } from '../../utils'
+import type { Instruction } from './models/Instruction'
+import { getTradeQuote, getTradeRate } from './swapperApi/getTradeQuote'
+import { getJupiterSwapInstructions } from './utils/helpers'
+
+export const jupiterApi: SwapperApi = {
+  getTradeQuote,
+  getTradeRate,
+  getUnsignedSolanaTransaction: async ({
+    tradeQuote,
+    from,
+    assertGetSolanaChainAdapter,
+    config,
+  }: GetUnsignedSolanaTransactionArgs): Promise<SolanaSignTx> => {
+    if (!isExecutableTradeQuote(tradeQuote)) throw Error('Unable to execute trade')
+
+    const jupiterUrl = config.REACT_APP_JUPITER_API_URL
+
+    const step = tradeQuote.steps[0]
+
+    if (!isExecutableTradeStep(step)) throw Error('Unable to execute step')
+
+    if (!tradeQuote.rawQuote) throw Error('Missing raw quote')
+
+    const contractAddress =
+      step.buyAsset.assetId === solAssetId
+        ? undefined
+        : fromAssetId(step.buyAsset.assetId).assetReference
+
+    const adapter = assertGetSolanaChainAdapter(step.sellAsset.chainId)
+
+    const isManualReceiveAddress = tradeQuote.receiveAddress !== from
+
+    if (isManualReceiveAddress && step.buyAsset.assetId === solAssetId) {
+      throw Error('Manual receive address is not supported for SOL')
+    }
+
+    const { instruction: createTokenAccountInstruction, destinationTokenAccount } =
+      contractAddress && isManualReceiveAddress
+        ? await adapter.createAssociatedTokenAccountInstruction({
+            from,
+            to: tradeQuote.receiveAddress,
+            tokenId: contractAddress,
+          })
+        : { instruction: undefined, destinationTokenAccount: undefined }
+
+    const maybeSwapResponse = await getJupiterSwapInstructions({
+      apiUrl: jupiterUrl,
+      fromAddress: from,
+      toAddress: isManualReceiveAddress ? destinationTokenAccount?.toString() : undefined,
```

Comment:
Straight but if manualreceive address, we add token ATA to this instead of account address